src/harmony/Input/Input.tsx

```diff
@@ -49,10 +60,25 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(
             outline,
             pointerEvents,
             forwardedRef,
+            autoFocus,
             ...rest
         },
         ref,
     ) => {
+        const [focused, setFocused] = useState(false);
+        const inputRef = useRef<HTMLInputElement>(null);
+        const forkedInputRef = useForkedRef(inputRef, ref);
+
+        useEffect(() => {
+            if (autoFocus && inputRef.current) {
+                inputRef.current.focus();
+                setFocused(true);
+            }
+            return () => {
+                setFocused(false);
+            };
+        }, [autoFocus, focused]);
```

Comment:
Effect depends on focused, setFocused is called inside effect. Looks like a cycle

Reply:
Yes, but the condition is preventing it from happening

Reply:
Even with condition it's going to fire twice on the first render.
And what is the reason for saving focused state in the hook?

Reply:
If the component has been mounted, the focus is not assigned when it is reopened